tests/Doctrine/Tests/Common/Reflection/StaticReflectionParserTest.php

```diff
@@ -101,7 +101,7 @@ public function testClassAnnotationOptimizedParsing($class, $classAnnotationOpti
     }
 
     /**
-     * @return string[]|bool[]
+     * @return array<array<bool|string>>
```

Comment:
This can be also annotated as
```suggestion
     * @return string[][]|bool[][]
```
That way not only static-analysis tools but PHPStorm will understand it too.

Reply:
In this case, this should be avoided: the signature you proposed translates to `array<array<bool>>|array<array<string>>`, which is a different type

Reply:
That is not the same though. The `array<...>` syntax allows for an array like this:
```
[
    ['foo', true],
    [true],
]
```

The "PHPStorm compatible" type does not properly allow for this.

Reply:
Oh, didn't know that nuance, thank you!